gauntlet/packages/gauntlet-serum-multisig/src/commands/approve.ts

```diff
@@ -0,0 +1,44 @@
+import { SolanaCommand, TransactionResponse } from '@chainlink/gauntlet-solana'
+import { PublicKey } from '@solana/web3.js'
+import { Result } from '@chainlink/gauntlet-core'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { CONTRACT_LIST, getContract } from '../lib/contracts'
+
+export default class Approve extends SolanaCommand {
+  static id = 'serum_multisig:approve'
```

Comment:
When is this going to be used?

Reply:
Reopening this, the whole purpose of the multisig command is validation. Here the users will approve proposals they don't know what they do.

If approval batching is required, it should be built within the multisig wrapper, so Gauntlet can make proper validations and inform the users about the actions they'll do.

Reply:
This is a quick workaround from an immediate need: without batching we need a ledger approval per feed per owner. This at least reduces it to a command (per 6-7 feeds) per owner.

It would be great to fix this in the multisig wrapper, but it's a larger job since we'd need to provide inputs for multiple invocations, verify them then pack that into a single tx. This is slightly tricky UX-wise too, since each command can take additional flags, for example `--secret` per `ocr2:accept_proposal:multisig`.

I think the multisig wrapper needs a general re-work to stop being so stateful too. It's very easy to accidentally call the code without a proposal (or worse yet, the code is faulty and fetches no proposal), which ends up creating new proposals rather than approving.

We've already had that happen on starknet: `proposalId` was missing in the implementation but the command was running successfully (and tests passing), creating new proposals over and over again: https://github.com/smartcontractkit/chainlink-starknet/commit/dacaae3d1dc4a3a21396fd7aad965288ff2f111e#diff-8257597e02534a1f29c2fcb789a25491779bd7ac33359412616550f5c422d83cR208